apis/v1/resourceSnapshot_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",shortName=rss,categories={fleet-workload}
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.metadata.generation`,name="Gen",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResourceSnapShot is used to store a snapshot of selected resources by a resource placement policy.
+// It is immutable. We may need to produce more than one resourceSnapshot for one ResourcePlacement to
+// get around the 1MB size limit of k8s objects.
+// Each snapshot must have a label "fleet.azure.com/snapshotGroup" with value as the snapshot index.
+// Each must have an annotation "fleet.azure.com/resource-hash" with value as the sha-256 hash
```

Comment:
should we put the resource hash as a field instead of annotation?

Reply:
The tricky thing is that this is not the value of this snapshot so I am not sure if this will be confusing for a field.